componentapiexample/src/main/java/net/gini/android/vision/component/camera/BaseCameraScreenHandler.java

```diff
@@ -0,0 +1,357 @@
+                GiniVisionFeatureConfiguration.buildNewConfiguration()
+                        .setDocumentImportEnabledFileTypes(
+                                DocumentImportEnabledFileTypes.PDF_AND_IMAGES)
+                        .setFileImportEnabled(true)
+                        .build();
+
+        if (savedInstanceState == null) {
+            final Intent intent = mActivity.getIntent();
+            if (isIntentActionViewOrSend(intent)) {
+                startGiniVisionLibraryForImportedFile(intent);
+            } else {
+                showCamera();
+            }
+        } else {
+            mCameraFragmentInterface = retainCameraFragment();
+        }
+    }
+
+    protected abstract void setUpActionBar();
+
+    private void showCamera() {
+        LOG.debug("Show the Camera Screen");
+        mCameraFragmentInterface = showCameraFragment();
+        // Delay notifying the coordinator to allow the camera fragment view to be created
+        new Handler().post(new Runnable() {
+            @Override
+            public void run() {
+                mGiniVisionCoordinator.onCameraStarted();
+            }
+        });
+    }
+
+    protected abstract CameraFragmentInterface showCameraFragment();
+
+    protected abstract CameraFragmentInterface retainCameraFragment();
+
+    private void configureLogging() {
+        final LoggerContext lc = (LoggerContext) LoggerFactory.getILoggerFactory();
+        lc.reset();
+
+        final PatternLayoutEncoder layoutEncoder = new PatternLayoutEncoder();
+        layoutEncoder.setContext(lc);
+        layoutEncoder.setPattern("%-5level %file:%line [%thread] - %msg%n");
+        layoutEncoder.start();
+
+        final LogcatAppender logcatAppender = new LogcatAppender();
+        logcatAppender.setContext(lc);
+        logcatAppender.setEncoder(layoutEncoder);
+        logcatAppender.start();
+
+        final ch.qos.logback.classic.Logger root =
+                (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
+        root.addAppender(logcatAppender);
+    }
+
+    private void setupGiniVisionCoordinator(final Activity activity) {
+        mGiniVisionCoordinator = GiniVisionCoordinator.createInstance(activity);
+        mGiniVisionCoordinator.setListener(new GiniVisionCoordinator.Listener() {
+            @Override
+            public void onShowOnboarding() {
+                showOnboarding();
+            }
+        });
+    }
+
+    private void showOnboarding() {
+        LOG.debug("Show the Onboarding Screen");
+        if (mMenu != null) {
+            mMenu.setGroupVisible(R.id.group, false);
+        }
+        mCameraFragmentInterface.hideInterface();
+        setTitlesForOnboarding();
+        showOnboardingFragment();
+    }
+
+    protected abstract void showOnboardingFragment();
+
+    protected abstract void setTitlesForOnboarding();
+
+    private void startGiniVisionLibraryForImportedFile(final Intent importedFileIntent) {
```

Comment:
Since `GiniVisionFileImport.createDocumentForImportedFile` method only acepts `@NonNull`intents, I think that it would be nice to add the annotation also in this method.